pkg/controller/operators/catalog/operator.go

```diff
@@ -1364,6 +1400,38 @@ func validateV1CRDCompatibility(dynamicClient dynamic.Interface, oldCRD *apiexte
 	return nil
 }
 
+// Validate all existing served versions against new CRD's validation (if changed)
+func validateV1Beta1CRDCompatibility(dynamicClient dynamic.Interface, oldCRD *apiextensionsv1beta1.CustomResourceDefinition, newCRD *apiextensionsv1beta1.CustomResourceDefinition) error {
+	logrus.Debugf("Comparing %#v to %#v", oldCRD.Spec.Validation, newCRD.Spec.Validation)
+	// If validation schema is unchanged, return right away
+	if reflect.DeepEqual(oldCRD.Spec.Validation, newCRD.Spec.Validation) {
```

Comment:
there are also validation schemas per version. You need both.

Reply:
This function verifies existing CRs against the new CRD validation and the comparison of the deep equals is just to avoid this check which is potentially expensive

Reply:
on second look - this should be removed entirely. Validation is an optional field, so if both are nil we may skip cr compatibility checks

Reply:
Removed